src/state.rs

```diff
@@ -0,0 +1,80 @@
+use std::time::{Duration, Instant};
+use crate::road::Road;
+use crate::pedestrian::Pedestrian;
+
+trait Vehicle {
+
+}
+
+trait State {
+
+    // const ROAD_LENGTH: f32;
+    // const CROSSINGS: Vec<Crossing>;
+
+
+    // fn get_vehicles(&self) -> &[dyn Vehicle];
+    fn timestamp(&self) -> Instant;
```

Comment:
I'm starting to think that my recommendation to use `Instant` and `Duration` for keeping track of the times was a mistake. I think ultimately we'd be better served just keeping it as a simple float with 0 as the initial time. I don't think that we actually gain anything by using this, having seen the issues during the day, and it might actually hurt us if/when we move to a unit-based system

Reply:
Yeah I tend to agree. I'm even wondering whether an integer time, say in milliseconds, would be more convenient for precise comparisons.